source/utilities.tex

```diff
@@ -8231,6 +8251,308 @@
+\pnum
+Let \tcode{G} be \tcode{remove_cvref_t<invoke_result_t<F, decltype(error())>>}.
+
+\pnum
+\constraints
+\tcode{is_copy_constructible_v<T>} is \tcode{true}.
+
+\pnum
+\mandates
+\tcode{G} is a specialization of \tcode{expected} and
+\tcode{is_same_v<G::value_type, T>} is \tcode{true}.
+
+\pnum
+\effects
+Equivalent to:
+\begin{codeblock}
+if (has_value())
+  return G(in_place, value());
+else
+  return invoke(std::forward<F>(f), error());
+\end{codeblock}
+\end{itemdescr}
+
+\indexlibrarymember{or_else}{expected}%
+\begin{itemdecl}
+template<class F> constexpr auto or_else(F&& f) &&;
+template<class F> constexpr auto or_else(F&& f) const &&;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{G} be
+\tcode{remove_cvref_t<invoke_result_t<F, decltype(std::move(error()))>>}.
+
+\pnum
+\constraints
+\tcode{is_move_constructible_v<T>} is \tcode{true}.
+
+\pnum
+\mandates
+\tcode{G} is a specialization of \tcode{expected} and
+\tcode{is_same_v<G::value_type, T>} is \tcode{true}.
+
+\pnum
+\effects
+Equivalent to:
+\begin{codeblock}
+if (has_value())
+  return G(in_place, std::move(value()));
+else
+  return invoke(std::forward<F>(f), std::move(error()));
+\end{codeblock}
+\end{itemdescr}
+
+\indexlibrarymember{transform}{expected}%
+\begin{itemdecl}
+template<class F> constexpr auto transform(F&& f) &;
+template<class F> constexpr auto transform(F&& f) const &;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{U} be
+\tcode{remove_cv_t<invoke_result_t<F, decltype(std::move(value()))>>}.
+
+\pnum
+\constraints
+\tcode{is_copy_constructible_v<E>} is \tcode{true}.
+
+\pnum
+\mandates
+U is a valid type for \tcode{expected}. If \tcode{is_void_v<U>} is
+\tcode{false}, the declaration
+\begin{codeblock}
+U u(invoke(std::forward<F>(f), value()));
+\end{codeblock}
+is well-formed.
+
+\pnum
+\effects
+\begin{itemize}
+\item
+If \tcode{has_value()} is \tcode{false}, returns
+\tcode{expected<U, E>(unexpect, error())}.
+\item
+Otherwise, if \tcode{is_void_v<U>} is \tcode{false}, returns an
+\tcode{expected<U, E>} object whose \exposid{has_val} member is \tcode{true}
+and \exposid{val} member is direct-non-list-initialized with
+\tcode{invoke(std::forward<F>(f), value())}.
+\item
+Otherwise, evaluates \tcode{invoke(std::forward<F>(f), value())} and then
+returns \tcode{expected<U, E>()}.
+\end{itemize}
+\end{itemdescr}
+
+\indexlibrarymember{transform}{expected}%
+\begin{itemdecl}
+template<class F> constexpr auto transform(F&& f) &&;
+template<class F> constexpr auto transform(F&& f) const &&;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{U} be
+\tcode{remove_cv_t<invoke_result_t<F, decltype(std::move(value()))>>}.
+
+\pnum
+\constraints
+\tcode{is_move_constructible_v<E>} is \tcode{true}.
+
+\pnum
+\mandates
+\tcode{U} is a valid value type for \tcode{expected}. If \tcode{is_void_v<U>} is
+\tcode{false}, the declaration
+\begin{codeblock}
+U u(invoke(std::forward<F>(f), std::move(value())));
+\end{codeblock}
+is well-formed.
```

Comment:
In the similar wording for `optional` we say "is well-formed for some invented variable `u`" here. I don't think it matters.

Reply:
Good idea, done!